runtime/parser/expression.go

```diff
@@ -965,12 +965,41 @@ func parseAttachExpressionRemainder(p *parser, token lexer.Token) (*ast.AttachEx
 	p.nextSemanticToken()
 
 	base, err := parseExpression(p, lowestBindingPower)
-
 	if err != nil {
 		return nil, err
 	}
 
-	return ast.NewAttachExpression(p.memoryGauge, base, attachment, token.StartPos), nil
+	p.skipSpaceAndComments()
+
+	var entitlements []*ast.NominalType
+	if p.isToken(p.current, lexer.TokenIdentifier, KeywordWith) {
+		// consume the `with` token
+		p.nextSemanticToken()
+
+		_, err = p.mustOne(lexer.TokenParenOpen)
+		if err != nil {
+			return nil, err
+		}
+
+		entitlements, _, err = parseNominalTypes(p, lexer.TokenParenClose, lexer.TokenComma)
+		for _, entitlement := range entitlements {
+			switch entitlement.Identifier.Identifier {
+			case KeywordAll, KeywordAccess, KeywordAccount, KeywordSelf:
+				return nil, p.syntaxError("unexpected non-nominal type: %s", entitlement)
+			}
```

Comment:
Can `rejectAccessKeywords` be used here?

Reply:
Yep!